File: packages/upay-php/examples/basic_usage.php
<?php

/**
 * Exemplo básico de uso do SDK Upay PHP
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Upay\UpayClient;

// Inicializar o cliente
$upay = new UpayClient(
    apiKey: "sua_api_key_aqui",
    baseUrl: "http://localhost:3001"  // Para desenvolvimento local
);

echo "=== Exemplo: Payment Links ===\n\n";

// Criar um link de pagamento
echo "1. Criando link de pagamento...\n";
$paymentLink = null;
$paymentLinkCreated = false;
try {
    $paymentLink = $upay->paymentLinks->create([
        'title' => 'Produto Premium',
        'amount' => 10000,  // R$ 100,00
        'description' => 'Descrição do produto',
        'status' => 'ACTIVE'
    ]);
    
    // Validar que o retorno é um array com as chaves necessárias
    if (!is_array($paymentLink) || !isset($paymentLink['id']) || !isset($paymentLink['slug'])) {
        echo "   Erro: Resposta inválida do servidor. Chaves esperadas: id, slug\n";
        echo "   Resposta recebida: " . json_encode($paymentLink) . "\n\n";
    } else {
        echo "   Link criado: {$paymentLink['id']}\n";
        echo "   Slug: {$paymentLink['slug']}\n";
        echo "   URL: " . $upay->paymentLinks->getCheckoutUrl($paymentLink['slug']) . "\n\n";
        $paymentLinkCreated = true;
    }
} catch (\Exception $e) {
    echo "   Erro ao criar link: {$e->getMessage()}\n";
    if ($e->getCode()) {
        echo "   Código: {$e->getCode()}\n";
    }
    echo "\n";
}

// Listar links
echo "2. Listando links de pagamento...\n";
try {
    $links = $upay->paymentLinks->list(['page' => 1, 'limit' => 5]);
    echo "   Total de links: " . ($links['pagination']['total'] ?? 0) . "\n";
    if (!empty($links['data'])) {
        echo "   Primeiro link: {$links['data'][0]['title']}\n\n";
    }
} catch (\Exception $e) {
    echo "   Erro: {$e->getMessage()}\n\n";
}

// Deletar o link de teste
if ($paymentLinkCreated && isset($paymentLink['id'])) {
    echo "3. Deletando link de teste...\n";
    try {
        $upay->paymentLinks->delete($paymentLink['id']);
        echo "   Link deletado com sucesso\n\n";
    } catch (\Exception $e) {
        echo "   Erro ao deletar: {$e->getMessage()}\n\n";
    }
}

echo "=== Exemplo: Transações ===\n\n";

// Listar transações
echo "1. Listando transações...\n";
$transactions = $upay->transactions->list(['page' => 1, 'limit' => 5]);
echo "   Total de transações: " . ($transactions['pagination']['total'] ?? 0) . "\n";
if (!empty($transactions['data'])) {
    $tx = $transactions['data'][0];
    $amount = number_format($tx['amountCents'] / 100, 2, ',', '.');
    echo "   Primeira transação: {$tx['product']} - R$ {$amount}\n\n";
}

echo "=== Exemplo: Produtos ===\n\n";

// Listar produtos
echo "1. Listando produtos...\n";
$products = $upay->products->list(['page' => 1, 'limit' => 5]);
echo "   Total de produtos: " . ($products['pagination']['total'] ?? 0) . "\n";
if (!empty($products['data'])) {
    $product = $products['data'][0];
    $price = number_format($product['priceCents'] / 100, 2, ',', '.');
    echo "   Primeiro produto: {$product['name']} - R$ {$price}\n\n";
}

echo "Todos os exemplos executados com sucesso!\n";

File: packages/upay-python/examples/basic_usage.py
"""
Exemplo básico de uso do SDK Upay Python
"""

from upay import UpayClient

# Inicializar o cliente
upay = UpayClient(
    api_key="sua_api_key_aqui",
    base_url="https://upay-sistema-api.onrender.com" 
)

def exemplo_payment_links():
    """Exemplo de uso de Payment Links"""
    print("=== Exemplo: Payment Links ===\n")
    
    # Criar um link de pagamento
    print("1. Criando link de pagamento...")
    payment_link = upay.payment_links.create({
        "title": "Produto Premium",
        "amount": 10000,  # R$ 100,00
        "description": "Descrição do produto",
        "status": "ACTIVE"
    })
    print(f"   Link criado: {payment_link.get('id')}")
    print(f"   Slug: {payment_link.get('slug')}")
    print(f"   URL: {upay.payment_links.get_checkout_url(payment_link['slug'])}\n")
    
    # Listar links
    print("2. Listando links de pagamento...")
    links = upay.payment_links.list(page=1, limit=5)
    print(f"   Total de links: {links['pagination'].get('total', 0)}")
    if links['data']:
        print(f"   Primeiro link: {links['data'][0].get('title')}\n")
    
    # Deletar o link de teste
    print("3. Deletando link de teste...")
    try:
        upay.payment_links.delete(payment_link['id'])
        print("   Link deletado com sucesso\n")
    except Exception as e:
        print(f"   Erro ao deletar: {e}\n")


def exemplo_transacoes():
    """Exemplo de uso de Transações"""
    print("=== Exemplo: Transações ===\n")
    
    # Listar transações
    print("1. Listando transações...")
    transactions = upay.transactions.list(page=1, limit=5)
    print(f"   Total de transações: {transactions['pagination'].get('total', 0)}")
    if transactions['data']:
        tx = transactions['data'][0]
        print(f"   Primeira transação: {tx.get('product')} - R$ {tx.get('amountCents', 0) / 100:.2f}\n")


def exemplo_produtos():
    """Exemplo de uso de Produtos"""
    print("=== Exemplo: Produtos ===\n")
    
    # Listar produtos
    print("1. Listando produtos...")
    products = upay.products.list(page=1, limit=5)
    print(f"   Total de produtos: {products['pagination'].get('total', 0)}")
    if products['data']:
        product = products['data'][0]
        print(f"   Primeiro produto: {product.get('name')} - R$ {product.get('priceCents', 0) / 100:.2f}\n")


def exemplo_clientes():
    """Exemplo de uso de Clientes"""
    print("=== Exemplo: Clientes ===\n")
    
    # Listar clientes
    print("1. Listando clientes...")
    clients = upay.clients.list(page=1, limit=5)
    print(f"   Total de clientes: {clients['pagination'].get('total', 0)}")
    if clients['data']:
        client = clients['data'][0]
        print(f"   Primeiro cliente: {client.get('name')} - {client.get('email')}\n")


def exemplo_cupons():
    """Exemplo de uso de Cupons"""
    print("=== Exemplo: Cupons ===\n")
    
    # Validar cupom
    print("1. Validando cupom...")
    try:
        validation = upay.coupons.validate(
            code="CUPOMTESTE",
            amount_cents=10000
        )
        if validation['valid']:
            print(f"   Cupom válido!")
            print(f"   Desconto: R$ {validation['discountCents'] / 100:.2f}")
            print(f"   Valor final: R$ {validation['finalAmountCents'] / 100:.2f}\n")
        else:
            print(f"   Cupom inválido: {validation.get('message')}\n")
    except Exception as e:
        print(f"   Erro: {e}\n")


if __name__ == "__main__":
    print("🧪 Testando SDK Upay Python\n")
    
    try:
        exemplo_payment_links()
        exemplo_transacoes()
        exemplo_produtos()
        exemplo_clientes()
        exemplo_cupons()
        
        print("✅ Todos os exemplos executados com sucesso!")
    except Exception as e:
        print(f"\n❌ Erro durante os testes: {e}")
        import traceback
        traceback.print_exc()
